backEnd/src/main/java/phylogenetictree/PhylogeneticNode.java

```diff
@@ -54,7 +54,8 @@ public PhylogeneticNode() {
      * @param distance The distance.
      * @param id       The id of this node, root being 0. Incrementing breath first.
      */
-    public PhylogeneticNode(final TreeNode node, final PhylogeneticNode parent, final double distance, final int id) {
+    public PhylogeneticNode(final TreeNode node, final PhylogeneticNode parent,
```

Comment:
Why is there a parent node needed now, compared to the previous version of this constructor without that parameter.